### PR TITLE
Uncouple ci building from e2e

### PR DESCRIPTION
**Pull request by mangelajo**
This will avoid the double-build of engine, routeagent and operator,
because they are built once for ci, and once again for e2e which
was not expected.

We start doing what documentation says, make compiles/builds, and e2e
just runs e2e with an existing build, README.md says:

```
To build `submariner-engine` and `submariner-route-agent` you can
trigger `make`, which will perform a Dapperized build of the components.

To run basic e2e tests you can trigger `make e2e` command.
```

### CODE REVIEW

**Comment by mangelajo** [comment]
Added build-engine script which was left out of the commit.

File: scripts/build [hunk 1]
```diff
@@ -4,7 +4,12 @@ set -e
...
-cd $(dirname $0)/..
-mkdir -p bin
-echo Building submariner-engine version $VERSION
-CGO_ENABLED=0 go build -ldflags "-X main.VERSION=$VERSION" -o bin/submariner-engine main.go
+cd $(dirname $0)/
+
+./build-engine
+./build-routeagent
+
+if [[ $5 = operator ]]; then
```
Comment (skitt) [inline scripts/build]: This needs to be handled differently now, we don’t have the arguments at this point.

Comment (mangelajo) [reply scripts/build]: right, there's my:

Error response from daemon: No such image: quay.io/submariner/submariner-operator:dev

thanks, I guess I can pass down the arguments to this script?

Comment (mangelajo) [reply scripts/build]: ok did that for now: https://github.com/submariner-io/submariner/pull/173/files#diff-799c33a5bd89478f978c187c2d3f9b79R10

**Comment by mangelajo** [comment]
CI worked. @tpantelis  ^

**Pull request merged**